### PR TITLE
feature/38299 implement pupil-login in tslib

### PR DESCRIPTION
**Pull request by GuyHarwood**
- [x] retain existing logic from `functions` app, without pupil-status push
- [x] rewrite in typescript under `tslib`
- [x] create function in `func-consumption`
- [x] only update db with login for live checks
- [x] only update db with login when state is correct
- [x] send login message to service bus queue
- [x] add all logins to pupilEvent table storage
- [x] BONUS: create distinction of old/new pupil API logic with separate controllers

### CODE REVIEW

File: tslib/src/functions/pupil-login/pupil-login.data.service.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  constructor () {
+    this.sqlService = new SqlService()
+  }
+
+  updateCheckWithLoginTimestamp (checkCode: string, loginDateTime: Moment): Promise<void> {
+    const sql = `UPDATE [mtc_admin].[check]
+               SET pupilLoginDate = @loginDate,
+                   checkStatus_id = (SELECT TOP 1 id FROM [mtc_admin].[checkStatus]
+                                    WHERE code = 'COL')
+               WHERE checkCode = @checkCode`
```
Comment (GuyHarwood) [inline tslib/src/functions/pupil-login/pupil-login.data.service.ts]: add guard to only update if pupilLoginDate is NULL

Comment (GuyHarwood) [reply tslib/src/functions/pupil-login/pupil-login.data.service.ts]: disable checkStatus update until pupil state work complete.  add PBI for it

File: func-consumption/pupil-login/function.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "disabled": false,
```
Comment (jon-shipley) [inline func-consumption/pupil-login/function.json]: This is a v1 setting.  Please remove.

File: tslib/src/functions/pupil-login/pupil-login.data.service.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  updateCheckWithLoginTimestamp (checkCode: string, loginDateTime: Date): Promise<void> {
+    const sql = `UPDATE [mtc_admin].[check]
+               SET pupilLoginDate = @loginDate,
+                   checkStatus_id = (SELECT TOP 1 id FROM [mtc_admin].[checkStatus]
+                                    WHERE code = 'COL')
+               WHERE checkCode = @checkCode
+               AND pupilLoginDate IS NULL
+               AND checkStatus_id = (SELECT TOP 1 id FROM [mtc_admin].[checkStatus]
+                WHERE code = 'NEW')`
```
Comment (jon-shipley) [inline tslib/src/functions/pupil-login/pupil-login.data.service.ts]: ✅

**Pull request merged**